src/models/stores/persistent-ui.ts

```diff
@@ -0,0 +1,268 @@
+  PersonalPublication, PlanningDocument, ProblemDocument,
+  ProblemPublication, SupportPublication } from "../document/document-types";
+import { UserModelType } from "./user";
+
+
+export const kSparrowAnnotationMode = "sparrow";
+
+// This generic model should work for both the problem tab, and the MyWork/ClassWork tabs
+// The StudentWorkspaces tab might work this way too. It has an open group which could be
+// stored in the openSubTab. And then it might have an open'd student document of the group
+// this would be the openDocument.
+export const UITabModel = types
+  .model("UITab", {
+    id: types.identifier,
+    openSubTab: types.maybe(types.string),
+    // The key of this map is the sub tab label
+    openDocuments: types.map(types.string),
+    openSecondaryDocuments: types.map(types.string)
+  });
+
+export const PersistentUIModel = types
+  .model("PersistentUI", {
+    dividerPosition: kDividerHalf,
+    activeNavTab: ENavTab.kProblems,
+    showAnnotations: true,
+    showTeacherContent: true,
+    showChatPanel: false,
+    tabs: types.map(UITabModel),
+    problemWorkspace: WorkspaceModel,
+    teacherPanelKey: types.maybe(types.string),
+  })
+  .volatile(self => ({
+    defaultLeftNavExpanded: false,
+    problemPath: ""
+  }))
+  .views((self) => ({
+    get navTabContentShown () {
+      return self.dividerPosition > kDividerMin;
+    },
+    get workspaceShown () {
+      return self.dividerPosition < kDividerMax;
+    },
+    get openSubTab () {
+      return self.tabs.get(self.activeNavTab)?.openSubTab;
+    },
+  }))
+  .views((self) => ({
+    // document key or section path for resource (left) document
+    get focusDocument () {
+      if (self.activeNavTab === ENavTab.kProblems || self.activeNavTab === ENavTab.kTeacherGuide) {
+        const facet = self.activeNavTab === ENavTab.kTeacherGuide ? ENavTab.kTeacherGuide : undefined;
+        return buildSectionPath(self.problemPath, self.openSubTab, facet);
+      } else {
+        const activeTabState = self.tabs.get(self.activeNavTab);
+        return self.openSubTab && activeTabState?.openDocuments.get(self.openSubTab);
+      }
+    },
+    get focusSecondaryDocument () {
+      if (self.activeNavTab === ENavTab.kProblems || self.activeNavTab === ENavTab.kTeacherGuide) {
+        const facet = self.activeNavTab === ENavTab.kTeacherGuide ? ENavTab.kTeacherGuide : undefined;
+        return buildSectionPath(self.problemPath, self.openSubTab, facet);
+      } else {
+        const activeTabState = self.tabs.get(self.activeNavTab);
+        return self.openSubTab && activeTabState?.openSecondaryDocuments.get(self.openSubTab);
+      }
+    },
+  }))
+  .actions((self) => {
+
+    const getTabState = (tab: string) => {
+      let tabState = self.tabs.get(tab);
+      if (!tabState) {
+        tabState = UITabModel.create({id: tab});
+        self.tabs.put(tabState);
+      }
+      return tabState;
+    };
+    return {
+      alert,
```

Comment:
I think this is a mistake. There is no local alert function in this file. I'm guessing this code isn't causing an error because this is picking up javascript's built in `alert` function.

Reply:
When `alert` was taken out, the build failed (see pic below) 
<img width="1425" alt="image" src="https://github.com/concord-consortium/collaborative-learning/assets/521934/03fe804d-33a8-4134-a473-0e13b1fcdf16">
So we decided to bring it back in.

Reply:
You should take it out and fix the reason the error happened. From the stacktrace it is in `document-workspace.tsx` 
Checking the code there is a reference to `persistentUI.alert(err.toString());` So that should be changed to `ui.alert(err.toString());`